errors.go

```diff
@@ -152,6 +152,34 @@ func ReplaceNativeMessage(e error, msg string) Error {
 	}
 }
 
+// AsType "casts" an input error as a target merry.Error such that the returned merry Error evaluates
+// to true when operating Is(newErr, targetErr). If e is nil, return nil. If targetType is nil,
+// A wrapped merry Error of e is returned. The key value pairs of the targetType is kept and then
+// appended with key value map from the input error. Stack trace of the original input error is maintained.
+func AsType(e error, targetType Error) (Error, bool) {
+	if targetType == nil {
+		return Wrap(e), false
+	}
+	switch e.(type) {
+	case nil:
+		return nil, false
+
+	case *merryErr:
+		output := targetType.Here()
+		output = output.WithValue(stack, Value(e, stack))
+
+		inputMap := Values(e)
+		for k, v := range inputMap {
+			output = output.WithValue(k, v)
+		}
+		return Append(output, e.Error()), true
+
+	default:
+		output := targetType.Here()
+		return Append(output, e.Error()), true
```

Comment:
If not a merry error we should probably use `ReplaceNativeMessage` instead, otherwise you can get major rollbar spam (@erismai -- ask one of us if this is unclear)

Reply:
My feeling is it should always be the user's responsibility to determine where ReplaceNativeMessage is needed. Not all projects (even all Helix projects) may be using Rollbar.